src/module/server-instance/ServerInstance.ts

```diff
@@ -0,0 +1,38 @@
+import {App} from "../../ManagementConsole";
+import {ServerInstanceCtrl} from "./ServerInstanceCtrl";
+import {ServerAddress} from "../../services/server/ServerAddress";
+import {IServerAddress} from "../../services/server/IServerAddress";
+
+const module: ng.IModule = App.module("managementConsole.server-instance", []);
+
+module.controller("ServerInstanceCtrl", ServerInstanceCtrl);
+
+// @ngInject
+module.config(($stateProvider: ng.ui.IStateProvider) => {
+    $stateProvider.state("server-instance", {
+        parent: "root",
+        url: "server-groups/:serverGroupName/:host/:server",
+        templateUrl: "module/server-instance/view/server-instance.html",
+        controller: ServerInstanceCtrl,
+        controllerAs: "ctrl",
+        resolve: {
+            coord: ["$q", "$stateParams", "serverGroupService", "jGroupsService", "serverService", ($q, $stateParams, serverGroupService, jGroupsService, serverService) => {
```

Comment:
We can prevent the compiler errors on $q by defining it as `$q: ng.IQService` in the parameters